### PR TITLE
API expansion and gitignore additions

### PR DESCRIPTION
**Pull request by geraldhumphries**

### CODE REVIEW

**Comment by sprice** [comment]
The expanded functions look great. Thanks!

I'm hesitant to include the changes to `.gitignore`. There's a lot of stuff in those changes that have no need in this repository (grunt for example, plus a lot more). I think it's best to have a project `.gitignore` cover what's needed for the project and nothing more. If you use Jetbrains and want certain Jetbrains files excluded from a git repository, use `.gitignore_global`. I don't use Jetbrains and don't want to see Jetbrains specific rules in some random open source project's `.gitignore`. That all being said, I'm happy to be persuaded that the change is useful if you have a strong argument for it.

**Comment by geraldhumphries** [comment]
Agreed, I've reverted the changes.

File: .gitignore [hunk 1]
```diff
@@ -1 +1 @@
-node_modules/
+node_modules
```
Comment (sprice) [inline .gitignore]: @geraldhumphries sorry for being picky but can you please remove all the changes to the `.gitignore`. After this I'll be happy to merge it in. Thanks for your work!

Comment (geraldhumphries) [inline .gitignore]: Sure, I didn't even notice the line wasn't the same. I'm actually going to cherry pick the main commit into a new branch and create a new pull request so we won't even see the .gitignore commits in master.

**Pull request closed without merging**